client/src/pages/Admin.js

```diff
@@ -1,43 +1,50 @@
+/* eslint-disable semi */
+/* eslint-disable no-undef */
+/* eslint-disable no-trailing-spaces */
+/* eslint-disable curly */
+/* eslint-disable linebreak-style */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 import React from "react";
 import { useState,useEffect } from "react";
 
 function AdminPage() {
-  const [pages, setPages] = useState([]);
+  const [pagesData, setPagesData] = useState([]);
 
-  async function fetchPagesAPI() {
-    const page = await fetch("/api/pages");
-    const response = await page.json();
-    setPages(response);
-	
+  async function fectPageTitles() {
+    try {
+      const getPageTitles = await fetch("/api/pages");
+      const allPageTitles = await getPageTitles.json();
+      return allPageTitles;
+    } catch (error) {
+      console.error(error);
+      return "Page TITLES  are not coming!! investigate why";
+    }
   }
- 
+
+  async function fectPagesData(pageTitle) {
+    try {
+      const getPagesData = await fetch(`api/pages/${pageTitle}`);
+      const pagesData = await getPagesData.json();
+      return pagesData;
+    } catch (error) {
+      console.error(error);
+      return "Pages DATA are not  coming!! investigate why";
+    }
+  }
+
   useEffect(() => {
-     fetchPagesAPI();
+    fectPageTitles()
+      .then((pagesTitleResults) => {
+        const data = pagesTitleResults.map((page) => fectPagesData(page.page_title));
```

Comment:
If you remember on ticket 24, we added a Promise.all()
That is missing here. So your code would be:
`const data = Promise.all(pagesTitleResults.map((page) => fectPagesData(page.page_title)));`

Reply:
> The Promise.all() static method takes an iterable of promises as input and returns a single [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). This returned promise fulfills when all of the input's promises fulfill (including when an empty iterable is passed), with an array of the fulfillment values. It rejects when any of the input's promises rejects, with this first rejection reason.